runtime/interpreter/value.go

```diff
@@ -17782,6 +17783,68 @@ func (v *CompositeValue) RemoveTypeKey(
 	return v.RemoveMember(interpreter, locationRange, attachmentMemberName(attachmentType))
 }
 
+func (v *CompositeValue) Iterator(interpreter *Interpreter) ValueIterator {
+	staticType := v.StaticType(interpreter)
+
+	switch staticType.(type) {
+	case InclusiveRangeStaticType:
+		startValue := v.GetField(interpreter, EmptyLocationRange, sema.InclusiveRangeTypeStartFieldName).(IntegerValue)
+		return &InclusiveRangeIterator{
+			rangeValue: v,
+			next:       startValue,
+		}
+
+	default:
+		// Must be caught in the checker.
+		panic(errors.NewUnreachableError())
+	}
+}
+
+type InclusiveRangeIterator struct {
+	rangeValue *CompositeValue
+	next       IntegerValue
+}
+
+var _ ValueIterator = &InclusiveRangeIterator{}
+
+func (i *InclusiveRangeIterator) Next(interpreter *Interpreter) Value {
+	valueToReturn := i.next
+
+	end := GetFieldAsIntegerValue(i.rangeValue, interpreter, EmptyLocationRange, sema.InclusiveRangeTypeEndFieldName)
+	step := GetFieldAsIntegerValue(i.rangeValue, interpreter, EmptyLocationRange, sema.InclusiveRangeTypeStepFieldName)
+
+	staticType := i.rangeValue.StaticType(interpreter)
+	var elementType StaticType
+	switch typ := staticType.(type) {
+	case InclusiveRangeStaticType:
+		elementType = typ.ElementType
+	default:
+		panic(errors.NewUnreachableError())
+	}
+
+	zeroValue := GetSmallIntegerValue(0, elementType)
+
+	// Ensure that valueToReturn is within the bounds.
+	if step.Less(interpreter, zeroValue, EmptyLocationRange) {
```

Comment:
This check doesn't need to be done every time the `next` is called.
Maybe introduce a constructor for the `InclusiveRangeIterator`, compute this inside the constructor and memoize it.

Initialization of the `startValue` can also be moved there instead of having it in the `Iterator()` method. So all initializations are in one place.

Reply:
Thanks, good suggestion. Done in d21296ea9.